### PR TITLE
fix(logging): fix service in log correlation

### PR DESCRIPTION
**Pull request by sileht**
# Feature

The service reported in logs can be wrong if set in the span itself.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### Testing
- [ ] Tests are run against all relevant library versions (or confirm that the relevant versions are already being tested in `tox.ini`)

### CODE REVIEW

File: ddtrace/tracer.py [hunk 1]
```diff
@@ -274,10 +274,14 @@ def get_log_correlation_context(self):
         if self.enabled:
             span = self.current_span()
 
+        service = config.service or ""
+        if span and span.service:
+            service = span.service
```
Comment (jd) [inline ddtrace/tracer.py:279]:
```suggestion
        if span and span.service:
            service = span.service
        else:
            service = config.service
```

**Review by jd — CHANGES_REQUESTED** [review]
LGTM since this implements the same logic we have in `start_span`.
However, I think you can simplify a bit the condition to find the service.

**Review by brettlangdon — COMMENTED** [review]
hey @sileht would you be able to add a release note as well:

```
pip install -U reno
reno new <some-short-note-slug>
```

This would be a `fixes:` note.

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by mabdinur — DISMISSED** [review]

**Comment by majorgreys** [comment]
@mergifyio backport 0.58

**Review by brettlangdon — APPROVED** [review]

File: ddtrace/tracer.py [hunk 2]
```diff
@@ -283,10 +283,15 @@ def get_log_correlation_context(self):
...
+        else:
+            service = config.service
+
         return {
             "trace_id": str(span.trace_id) if span else "0",
             "span_id": str(span.span_id) if span else "0",
-            "service": config.service or "",
+            "service": service,
```
Comment (majorgreys) [inline ddtrace/tracer.py:294]: We want to retain an empty string as the default as we do for `version` and `env`:

```suggestion
            "service": service or "",
```

**Review by majorgreys — DISMISSED** [review]

**Review by brettlangdon — DISMISSED** [review]

**Review by mabdinur — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**